src/Common/test/Certificates.Test/ConfigureCertificateOptionsTest.cs

```diff
@@ -17,176 +18,287 @@ public sealed class ConfigureCertificateOptionsTest
+        configureOptions.Configure(certificateName, options);
 
         options.Certificate.Should().BeNull();
     }
 
-    [Fact]
-    public void ConfigureCertificateOptions_BadPath_NoCertificate()
+    [Theory]
+    [InlineData("")]
+    [InlineData(CertificateName)]
+    public void ConfigureCertificateOptions_BadPath_NoCertificate(string certificateName)
     {
-        IConfigurationRoot configurationRoot = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string?>
+        IConfiguration configuration = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string?>
         {
-            [$"{CertificateOptions.ConfigurationKeyPrefix}:{CertificateName}:certificateFilePath"] = "does-not-exist.crt"
+            [$"{GetConfigurationKey(certificateName, "CertificateFilePath")}"] = "does-not-exist.crt"
         }).Build();
 
-        var configureOptions = new ConfigureCertificateOptions(configurationRoot);
+        var configureOptions = new ConfigureCertificateOptions(configuration);
 
         var options = new CertificateOptions();
 
-        configureOptions.Configure(CertificateName, options);
+        configureOptions.Configure(certificateName, options);
 
         options.Certificate.Should().BeNull();
     }
 
-    [Fact]
-    public void ConfigureCertificateOptions_EmptyFile_Crashes()
+    [Theory]
+    [InlineData("")]
+    [InlineData(CertificateName)]
+    public void ConfigureCertificateOptions_EmptyFile_Crash_logged(string certificateName)
     {
-        IConfigurationRoot configurationRoot = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string?>
+        CapturingLoggerProvider loggerProvider = new();
+        using var loggerFactory = new LoggerFactory([loggerProvider]);
+
+        IConfiguration configuration = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string?>
         {
-            [$"{CertificateOptions.ConfigurationKeyPrefix}:{CertificateName}:certificateFilePath"] = "empty.crt"
+            [$"{GetConfigurationKey(certificateName, "CertificateFilePath")}"] = "empty.crt"
         }).Build();
 
-        var configureOptions = new ConfigureCertificateOptions(configurationRoot);
+        var configureOptions = new ConfigureCertificateOptions(configuration, loggerFactory.CreateLogger<ConfigureCertificateOptions>());
         var options = new CertificateOptions();
 
-        Action action = () => configureOptions.Configure(CertificateName, options);
-        action.Should().Throw<CryptographicException>();
-
+        configureOptions.Configure(certificateName, options);
         options.Certificate.Should().BeNull();
+
+        loggerProvider.GetAll().Should()
+            .Contain(
+                $"DBUG Steeltoe.Common.Certificates.ConfigureCertificateOptions: CryptographicException while parsing certificate for '{certificateName}' from 'empty.crt'. Will retry on next reload.");
```

Comment:
The Resharper reformatter isn't perfect. It sometimes inserts too many like breaks. In this case, it needs a bit of help. Move the `.Contain(` after the end of the first line and reformat. That leaves us with two lines instead of three.

![image](https://github.com/user-attachments/assets/984afb9f-b35f-4beb-aa57-472a0dedaafc)